src/uwp/Host/UWPHost/HostEnvironment.cpp

```diff
@@ -0,0 +1,246 @@
+#include "HostEnvironment.h"
+
+
+HRESULT HostEnvironment::TryLoadCoreCLR() {
+
+    m_coreCLRModule = ::LoadLibraryEx(coreCLRDll, NULL, 0);
+    if (!m_coreCLRModule) {
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
+
+    // Pin the module - CoreCLR.dll does not support being unloaded.
+    // N.B.: HostEnvironment is not calling ::FreeLibrary on this loaded module since we're pinning it anyway.
+    // If unloading CoreCLR is ever supported and the pinning below is to be removed please make sure that 
+    // the ::FreeLibrary is called appropriately.
+    HMODULE dummy_coreCLRModule;
+    if (!::GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_PIN, coreCLRDll, &dummy_coreCLRModule)) 
+    {
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
+
+    if (!GetModuleFileNameW(m_coreCLRModule, m_coreCLRInstallDirectory, MAX_PATH))
+    {
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
+    // Find out where the last backslash is, and trim the path to get the directory containing coreclr.dll
+    wchar_t *lastBackSlash = wcsrchr(m_coreCLRInstallDirectory, L'\\');
+    if (lastBackSlash != nullptr)
+    {
+        *(lastBackSlash+1) = L'\0';
+    }
+    else
+    {
+        memset(m_coreCLRInstallDirectory, 0, MAX_PATH * sizeof(wchar_t)); 
+        return E_FAIL;
+    }
+
+    return S_OK;
+}
+
+
+HRESULT HostEnvironment::Initialize()
+{
+    HRESULT hr = S_OK;
+    // 
+    // Get the package root. 
+    // This is also our check for the App to be an AppX app.
+    // TODO: Not all AppX apps are UWP apps. Add validation for UWP apps.
+    //
+    UINT32 length = 0;
+    LONG rc = GetCurrentPackagePath(&length, NULL);
+    if (rc != ERROR_INSUFFICIENT_BUFFER)
+    {
+        return HRESULT_FROM_WIN32(rc);
+    }
+
+    m_packageRoot = (PWSTR) malloc(length * sizeof(*m_packageRoot));
+    if (m_packageRoot == NULL)
+    {
+        return E_FAIL;
+    }
+
+    // Get the actual path to the package root from the path to the current module.
+    rc = GetCurrentPackagePath(&length, m_packageRoot);
+    IfFailWin32Ret(rc);
+
+    m_fIsAppXPackage = true;
+
+    //
+    // Query the package full name
+    //
+    length = 0;
+    rc = GetCurrentPackageFullName(&length, NULL);
+    if (rc != ERROR_INSUFFICIENT_BUFFER)
+    {
+        return HRESULT_FROM_WIN32(rc);
+    }
+
+    m_currentPackageFullName = (PWSTR) malloc(length * sizeof(*m_currentPackageFullName));
+    if (m_currentPackageFullName == NULL)
+    {
+        return HRESULT_FROM_WIN32(rc);
+    }
+
+    rc = GetCurrentPackageFullName(&length, m_currentPackageFullName);
+    IfFailWin32Ret(rc);
+    
+    // Try to load CoreCLR
+    IfFailRet(TryLoadCoreCLR());
+
+
+    PACKAGE_INFO_REFERENCE packageInfoReference = {0};
+    rc = OpenPackageInfoByFullName(m_currentPackageFullName, 0 , &packageInfoReference);
+    IfFailWin32Ret(rc);
+
+    length = 0;
+    rc = GetPackageInfo(packageInfoReference, PACKAGE_FILTER_DIRECT, &length, NULL, &m_dependentPackagesCount);
```

Comment:
Can you please update the flags to pull in the optional packages flag I talked to you about last week?